src/main/java/Rabbit/util/Parser.java

```diff
@@ -0,0 +1,224 @@
+package Rabbit.util;
+
+import java.lang.module.FindException;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import Rabbit.Task.Task;
+import Rabbit.Task.Event;
+import Rabbit.Task.Deadline;
+import Rabbit.Task.Todo;
+import Rabbit.RabbitException.AddToListException;
+import Rabbit.RabbitException.MarkUnmarkException;
+import Rabbit.RabbitException.DeleteException;
+import Rabbit.RabbitException.ImportDataException;
+import Rabbit.RabbitException.FindFormatException;
+
+public class Parser {
+    public Parser(){};
+
+    /**
+     * Parses the lines in data.txt to tasks that
+     * are added into the list.
+     *
+     * @param input lines of tasks from data.txt.
+     * @return the task to be added.
+     * @throws ImportDataException
+     */
+    public static Task parseImport(String input) throws ImportDataException {
+        char type = input.charAt(0);
```

Comment:
I really like how you catch errors in this method! Do you think that perhaps Java String [split method with regex ](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String)) instead of using `substring` might help make the method briefer?